tools/clang/lib/SPIRV/PervertexInputVisitor.cpp

```diff
@@ -0,0 +1,443 @@
+//===--- PervertexInputVisitor.cpp ---- PerVertex Input Visitor --*- C++ -*-==//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PervertexInputVisitor.h"
+#include "clang/SPIRV/AstTypeProbe.h"
+#include "clang/SPIRV/SpirvFunction.h"
+#include "clang/SPIRV/SpirvType.h"
+
+#include <stack>
+
+namespace clang {
+namespace spirv {
+
+///< After expanding inputs, location need reassignment.
```

Comment:
I mentioned this in a comment on a test. Can you explain why we need reassignment for the location? I worry this will make it very difficult to match the vertex shader outputs and pixel shader inputs.

Reply:
This case has been removed